x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/IsNullProcessor.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.expression.predicate;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xpack.sql.expression.gen.processor.Processor;
+
+import java.io.IOException;
+
+public class IsNullProcessor implements Processor {
```

Comment:
As we have two very similar processors, it makes sense to aggregate the isnull/isnotnull processors into a single processor class which uses an enum to differentiate between the two (as we do with the rest of other operators).
This makes things consistent and keeps the serialization name explosion to a minimum.
It's also a good reason to keep this a 6.6 feature and not backport it to 6.5 (since it's an enhancement not really a bug).

Reply:
I've tested this PR for this bug - https://github.com/elastic/elasticsearch/issues/35171 and it's no longer reproduceable... might be a good reason to backport it to 6.5 @costin ?

Reply:
If that's the case and the change isn't too big, it's worth backporting yes.